### PR TITLE
Subnet logconfig prepare for 3.0.0

### PR DESCRIPTION
**Pull request by slevenick**
This moves log_config to GA, deprecates `enable_flow_logs`, infers that `enable_flow_logs` is true if `log_config` is present and `enable_flow_logs` is unspecified. Also allows for `log_config` to be updated without recreating the subnetwork resource.

Prepares to remove `enable_flow_logs` in 3.0.0 in favor of inferring `log_config.enable` 

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
`compute`: `log_config` is now available in GA for `google_compute_subnetwork`
```

```release-note:deprecation
`compute`: deprecated `enable_flow_logs` on `google_compute_subnetwork`. The presence of the `log_config` block signals that flow logs are enabled for a subnetwork
```

### CODE REVIEW

File: products/compute/terraform.yaml [hunk 1]
```diff
@@ -244,8 +244,6 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         sensitive: true
       id: !ruby/object:Overrides::Terraform::PropertyOverride
         exclude: true
-      logConfig: !ruby/object:Overrides::Terraform::PropertyOverride
```
Comment (rileykarson) [inline products/compute/terraform.yaml]: This seems unrelated?

File: products/compute/terraform.yaml [hunk 2]
```diff
@@ -1677,8 +1680,6 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         input: false
         default_from_api: true
         custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.erb'
-      logConfig: !ruby/object:Overrides::Terraform::PropertyOverride
```
Comment (rileykarson) [inline products/compute/terraform.yaml]: Are we able to keep the field O+C? I'm a little hesitant to start catching drift out of the blue, especially on our last `2.X` release. However, it's beta-only so it's allowed imo.

File: products/compute/terraform.yaml (hunk 1 above)
Comment (slevenick) [reply products/compute/terraform.yaml]: Whoops! Must have searched on `logConfig` and gotten mixed up!
